From 2bf5706f73b06ce89a85aa37dcd0e9ded8d59c03 Mon Sep 17 00:00:00 2001
From: Oliver Smith <ollieparanoid@postmarketos.org>
Date: Sun, 2 May 2021 13:20:46 +0200
Subject: [PATCH] [mobile] add option builtinVirtualKeyboard

Allow to always hide the built-in qtvirtualkeyboard to support using a
different keyboard.

Related: https://gitlab.com/postmarketOS/postmarketos-ondev/-/issues/47
---
 modules/mobile/Config.cpp  | 2 ++
 modules/mobile/Config.h    | 9 +++++++++
 modules/mobile/mobile.conf | 4 ++++
 modules/mobile/mobile.qml  | 1 +
 4 files changed, 16 insertions(+)

diff --git a/modules/mobile/Config.cpp b/modules/mobile/Config.cpp
index 772f8d4..e6078b7 100644
--- a/modules/mobile/Config.cpp
+++ b/modules/mobile/Config.cpp
@@ -26,6 +26,8 @@ Config::setConfigurationMap( const QVariantMap& cfgMap )
     m_version = getString( cfgMap, "version", "(unknown)" );
     m_username = getString( cfgMap, "username", "user" );
 
+    m_builtinVirtualKeyboard = getBool( cfgMap, "builtinVirtualKeyboard", true );
+
     m_featureSshd = getBool( cfgMap, "featureSshd", true );
     m_featureFsType = getBool( cfgMap, "featureFsType", false );
 
diff --git a/modules/mobile/Config.h b/modules/mobile/Config.h
index ea304ec..5c39c77 100644
--- a/modules/mobile/Config.h
+++ b/modules/mobile/Config.h
@@ -10,6 +10,9 @@
 class Config : public QObject
 {
     Q_OBJECT
+    /* installer UI */
+    Q_PROPERTY( bool builtinVirtualKeyboard READ builtinVirtualKeyboard CONSTANT FINAL )
+
     /* welcome */
     Q_PROPERTY( QString osName READ osName CONSTANT FINAL )
     Q_PROPERTY( QString arch READ arch CONSTANT FINAL )
@@ -58,6 +61,9 @@ public:
     void setConfigurationMap( const QVariantMap& );
     Calamares::JobList createJobs();
 
+    /* installer UI */
+    bool builtinVirtualKeyboard() { return m_builtinVirtualKeyboard; }
+
     /* welcome */
     QString osName() const { return m_osName; }
     QString arch() const { return m_arch; }
@@ -117,6 +123,9 @@ public:
     QString cmdSshdUseradd() const { return m_cmdSshdUseradd; }
 
 private:
+    /* installer UI */
+    bool m_builtinVirtualKeyboard;
+
     /* welcome */
     QString m_osName;
     QString m_arch;
diff --git a/modules/mobile/mobile.conf b/modules/mobile/mobile.conf
index dbae56c..7b9dbd5 100644
--- a/modules/mobile/mobile.conf
+++ b/modules/mobile/mobile.conf
@@ -64,6 +64,10 @@
 ## this gets used without asking the user.
 # defaultFs: ext4
 
+## Start Qt's virtual keyboard within the mobile module. Disable if you bring
+## your own virtual keyboard (e.g. svkbd).
+# builtinVirtualKeyboard: true
+
 #######
 ### Commands running in the installer OS
 #######
diff --git a/modules/mobile/mobile.qml b/modules/mobile/mobile.qml
index 8411a39..3821345 100644
--- a/modules/mobile/mobile.qml
+++ b/modules/mobile/mobile.qml
@@ -142,6 +142,7 @@ Page
     InputPanel {
         id: inputPanel
         y: Qt.inputMethod.visible ? parent.height - inputPanel.height : parent.height
+        visible: config.builtinVirtualKeyboard
         anchors.left: parent.left
         anchors.right: parent.right
     }
-- 
2.26.3

